client/app/match/match.js

```diff
@@ -33,82 +25,89 @@ angular.module( 'dinnerDaddy.match', ['dinnerDaddy.services'] )
-      Match.checkMatch($scope.session, $scope.currRestaurant)
-      .then(function (result) {
-        if (result) {
-          Socket.emit( 'foundMatch', { sessionName: $scope.session.sessionName, restaurant: $scope.currRestaurant } );
-        } else {
-          loadNextRestaurant(); 
-        }
-      });
+      Match.checkMatch($scope.currentSession.id, currRestaurantIndex)
+        .then(function (result) {
+          if (result !== false) {
+            Socket.emit('foundMatch', {sessionName: $rootScope.currentSession.sessionName, restaurant: currRestaurantIndex, sessionId: $rootScope.currentSession.id, matched: $scope.currRestaurant});
+            $rootScope.matched = $scope.currRestaurant;
+            $window.localStorage.setItem('matched', JSON.stringify($rootScope.matched));
+          } else {
+            loadNextRestaurant(); 
+          }
+        });
     });
   }
+
   $scope.no = function() {
-    // debugger;
-    console.log('clicked on NO');
-    Match.sendVote($scope.session.sessionName, $scope.user.name, currRestaurantIndex, false);
-    console.log('the scope contains: ', $scope);
+    Match.sendVote($rootScope.currentSession.sessionName, $scope.user.name, currRestaurantIndex, false, $rootScope.currentSession.id);
     loadNextRestaurant();
   }
+
+  // Listen for the signal to redirect to a 'match found' page.
+  Socket.on('matchRedirect', function (id) {
+    // id refers to the id of the movie that the group matched on
+    Match.matchRedirect(id);
+  });
+
+  // Listen for the signal to reset matched on rootScope
+  Socket.on('setMatched', function (data) {
+    // Set matched on rootScope to matched restaurant
+    $rootScope.matched = data;
+    // Set the image url from small to large
+    var currentImageURL = data.image_url;
+    $rootScope.currRestaurantImageHD = currentImageURL.slice(0,currentImageURL.length-6) + 'l.jpg'; 
+    $window.localStorage.setItem('matched', JSON.stringify($rootScope.matched));
+  });
 })
-.factory( 'Match', function($http, $location) {
+
+.factory('Match', function ($http, $location) {
   return {
-    sendVote: function(sessionName, username, movieID, vote) {
-      console.log('sending vote....');
-      return $http.post( // returns a promise; if you want to make use of a callback simply use .then on the return value.
-        '/api/votes', // expect this endpoint to take a json object
-                                      // with sessionID and userID
-                                      // OR sessionuserID
-                                      // AND movieID
-                                      // AND vote (boolean true/false where true is yes and false is no)
-        { sessionName: sessionName, username: username, movie_id: movieID, vote: vote})
-      .then(function (response) { // if the promise is resolved
-        console.log('promise resolved in sendvote, resp is: ', response);
+    sendVote: function (sessionName, username, restaurantId, vote, sessionId) {
```

Comment:
remove any unnecessary variables here--do we need to send sessionName and username to the server? Is this based on the legacy code we have in terms of how the votes table works? If so that's fine.